authorization/team/service/team_service.go

```diff
@@ -0,0 +1,178 @@
+			return errors.NewBadParameterErrorFromString("spaceID", spaceID, "invalid space ID specified")
+		}
+
+		// Confirm that the resource is a space
+		if space.ResourceType.Name != authorization.ResourceTypeSpace {
+			return errors.NewBadParameterErrorFromString("spaceID", spaceID, "space ID specified is not a space resource")
+		}
+
+		// Create the permission service
+		permService := permissionservice.NewPermissionService(tr)
+
+		// Confirm that the user has the 'manage' scope for the space
+		scope, err := permService.HasScope(ctx, identityID, spaceID, authorization.ManageTeamsInSpaceScope)
+		if err != nil {
+			return errors.NewInternalError(ctx, err)
+		}
+
+		if !scope {
+			return errors.NewForbiddenError(fmt.Sprintf("user requires %s scope for the space to be able to create new teams", authorization.ManageTeamsInSpaceScope))
+		}
+
+		// Lookup the team resource type
+		resourceType, err := tr.ResourceTypeRepository().Lookup(ctx, authorization.IdentityResourceTypeTeam)
+		if err != nil {
+			return err
+		}
+
+		// Create the team resource
+		res := &resource.Resource{
+			Name:             teamName,
+			ResourceType:     *resourceType,
+			ResourceTypeID:   resourceType.ResourceTypeID,
+			ParentResourceID: &spaceID,
+		}
+
+		err = tr.ResourceRepository().Create(ctx, res)
+		if err != nil {
+			return errors.NewInternalError(ctx, err)
+		}
+
+		// Create the team identity
+		teamIdentity := &account.Identity{
+			IdentityResourceID: sql.NullString{res.ResourceID, true},
+		}
+
+		err = tr.Identities().Create(ctx, teamIdentity)
+		if err != nil {
+			return errors.NewInternalError(ctx, err)
+		}
+
+		teamID = teamIdentity.ID
+
+		log.Debug(ctx, map[string]interface{}{
+			"team_id": teamID.String(),
+		}, "team created")
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &teamID, nil
+}
+
+// Returns an array of all team identities within a space
+func (s *teamServiceImpl) ListTeamsInSpace(ctx context.Context, identityID uuid.UUID, spaceID string) ([]account.Identity, error) {
+	// Confirm that the specified spaceID is valid
+	space, err := s.repo.ResourceRepository().Load(ctx, spaceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Confirm that the resource is actually a space
+	if space.ResourceType.Name != authorization.ResourceTypeSpace {
+		return nil, errors.NewBadParameterErrorFromString("spaceID", spaceID, "invalid space ID specified - resource is not a space")
+	}
+
+	// Create the permission service
+	permService := permissionservice.NewPermissionService(s.repo)
+
+	// Confirm the user has the necessary privileges to list the teams in this space
+	scope, err := permService.HasScope(ctx, identityID, spaceID, authorization.ViewTeamsInSpaceScope)
+	if err != nil {
+		return nil, errors.NewInternalError(ctx, err)
+	}
+
+	if !scope {
+		return nil, errors.NewUnauthorizedError(fmt.Sprintf("user requires %s scope for the space to be able to view teams", authorization.ViewTeamsInSpaceScope))
```

Comment:
I've added ForbiddenError to the response codes, and adjusted the controller test to suit.  Rather than modifying the `HasScope()` method in PermissionService (because I still think it may have use cases) I've introduced another method called `RequireScope()`, which will return a `ForbiddenError` if the user does not have the scope.  I've then used this in the team service to simplify the code.